source/remoteClient/secureDesktop.py

```diff
@@ -0,0 +1,240 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2015-2025 NV Access Limited, Christopher Toth, Tyler Spivey, Babbage B.V., David Sexton and others.
+# This file is covered by the GNU General Public License.
+# See the file COPYING for more details.
+
+"""Secure desktop support for NVDA Remote.
+
+This module handles the transition between regular and secure desktop sessions in Windows,
+maintaining remote connections across these transitions. It manages the creation of local
+relay servers, connection bridging, and IPC (Inter-Process Communication) between the
+regular and secure desktop instances of NVDA.
+
+The secure desktop is a special Windows session used for UAC prompts and login screens
+that runs in an isolated environment for security. This module ensures NVDA Remote
+connections persist when entering and leaving this secure environment.
+"""
+
+import json
+import socket
+import threading
+import uuid
+from pathlib import Path
+from typing import Any, Optional
+
+import shlobj
+from logHandler import log
+from winAPI.secureDesktop import post_secureDesktopStateChange
+
+from . import bridge, server
+from .connectionInfo import ConnectionInfo, ConnectionMode
+from .protocol import RemoteMessageType
+from .serializer import JSONSerializer
+from .session import SlaveSession
+from .transport import RelayTransport
+
+
+def getProgramDataTempPath() -> Path:
+	"""Get the system's program data temp directory path."""
+	if hasattr(shlobj, "SHGetKnownFolderPath"):
+		return Path(shlobj.SHGetKnownFolderPath(shlobj.FolderId.PROGRAM_DATA)) / "temp"
+	return Path(shlobj.SHGetFolderPath(0, shlobj.CSIDL_COMMON_APPDATA)) / "temp"
```

Comment:
`shlobj.SHGetFolderPath` was removed in NVDA 2022.1. Only use `SHGetKnownFolderPath.
```suggestion
	return Path(shlobj.SHGetKnownFolderPath(shlobj.FolderId.PROGRAM_DATA)) / "temp"
```